### PR TITLE
PLT-7785: Slash commands can be issued to a channel in a team without it

### PR DESCRIPTION
**Pull request by MusikPolice**
#### Summary
Ensured that specified channel is a part of specified team prior to executing the slash command.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-7785

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: model/client4.go [hunk 1]
```diff
@@ -2807,8 +2807,12 @@ func (c *Client4) ListCommands(teamId string, customOnly bool) ([]*Command, *Res
 }
 
 // ExecuteCommand executes a given command.
-func (c *Client4) ExecuteCommand(channelId, command string) (*CommandResponse, *Response) {
-	commandArgs := &CommandArgs{ChannelId: channelId, Command: command}
+func (c *Client4) ExecuteCommand(teamId, channelId, command string) (*CommandResponse, *Response) {
```
Comment (ccbrown) [inline model/client4.go]: This change isn't really necessary is it? I think we aim to keep the ClientX APIs stable.

File: api4/command.go [hunk 2]
```diff
@@ -212,9 +212,13 @@ func executeCommand(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// DMs don't have teams, so get the team id from the channel if not specified
```
Comment (crspeller) [inline api4/command.go]: Now that I think about it. Why even bother letting the user specify TeamId? Just do the `commandArgs.TeamId = channel.TeamId` every time.

File: model/client4.go (hunk 1 above)
Comment (jwilander) [reply model/client4.go]: Yes, sorry @MusikPolice I meant to mention that but forgot after we went down the rabbit hole. We should create a new function `ExecuteCommandWithTeamId` if we need to specify the team ID

File: api4/command.go (hunk 2 above)
Comment (MusikPolice) [reply api4/command.go]: What, if anything, do we do about `CommandArgs` with this change? Do we need to deprecate the `TeamId` field in some way?

**Pull request merged**